### PR TITLE
Allow tuning of dupe adds in Hammer

### PR DESCRIPTION
**Pull request by AlCutter**
Currently, the hammer chooses whether to add a new or a duplicate submission with roughly 50% hardcoded probability, this change allows the user to set their own probability.

### Checklist

- [X] I have updated the [CHANGELOG](CHANGELOG.md).
  - Adjust the draft version number according to [semantic versioning](https://semver.org/) rules.
- [ ] I have updated [documentation](docs/) accordingly.

### CODE REVIEW

File: trillian/integration/hammer.go [hunk 1]
```diff
@@ -141,6 +141,10 @@ type HammerConfig struct {
 	MaxRetryDuration time.Duration
 	// RequestDeadline indicates the deadline to set on each request to the log.
 	RequestDeadline time.Duration
+	// DuplicateChance sets the probability of attempting to add a duplicate when
+	// calling add[-pre]-chain (as the N in 1-in-N). Set to 0 to disable sending
+	// duplicates.
+	DuplicateChance int
```
Comment (Martin2112) [inline trillian/integration/hammer.go]: It's a slight shame this has to be different to the existing `Bias` mechanisms but they seem to be handler specific. 

Not sure if DuplicateFraction would be easier to understand and then have 0-1 random values.

File: trillian/integration/hammer.go [hunk 2]
```diff
@@ -489,13 +492,21 @@ func (s *hammerState) addChainInvalid(ctx context.Context) error {
 	return nil
 }
 
+// chooseAddType determines whether to add a new or pre-existing cert.
```
Comment (Martin2112) [inline trillian/integration/hammer.go]: Comment / function name mismatch.

File: trillian/integration/hammer.go [hunk 3]
```diff
@@ -489,13 +492,21 @@ func (s *hammerState) addChainInvalid(ctx context.Context) error {
...
+func (s *hammerState) chooseCertToAdd() Choice {
+	if s.cfg.DuplicateChance > 0 && rand.Intn(s.cfg.DuplicateChance) == 0 {
+		// TODO(drysdale): restore LastCert as an option
+		return FirstCert
```
Comment (Martin2112) [inline trillian/integration/hammer.go]: Can `FirstCert` be nil when we get here?

File: trillian/integration/hammer.go (hunk 1 above)
Comment (AlCutter) [reply trillian/integration/hammer.go]: The 1-in-N thing already exists in the codebase so it seemed better to stick with that.

Comment (Martin2112) [reply trillian/integration/hammer.go]: OK.

File: trillian/integration/hammer.go (hunk 2 above)
Comment (AlCutter) [reply trillian/integration/hammer.go]: whoops, fixed!

File: trillian/integration/hammer.go (hunk 3 above)
Comment (AlCutter) [reply trillian/integration/hammer.go]: Don't think so - `FirstCert` is a `const`...

Comment (Martin2112) [reply trillian/integration/hammer.go]: OK I see it now, thought it might have been holding onto a generated cert.

Comment (AlCutter) [reply trillian/integration/hammer.go]: Ah, yes - it does do that elsewhere in pre-existing code.

**Pull request merged**